part2/stages/Functions/install-main

```diff
@@ -591,7 +594,18 @@ create_efi_boot_entries()
 
 copy_to_esp()
 {
-    local ESP=$(sfdisk -l -q | grep "EFI System" | awk '{ print $1 }')
+    # TARGET_DISK not defined in upgrades; locate our
+    # installation disk manually.
+    if [ -z "${TARGET_DISK}" ]; then
+        local part="$(vgs xenclient --noheadings -o pv_name | sed 's/\s*//g' | sed 's|^/dev/||g')"
+
+        # get disk from partition (e.g. sda2 --> sda). using
+        # readlink instead of simply stripping off trailing
+        # numbers to cover other disk formats (e.g. nvme)
```

Comment:
The comment makes it sound like it's readlink resolving to the disk, but it's really the sysfs directory hierarchy.  Maybe say "Use readlink to get the /sys/device entry and then use the parent directory as the disk name"